Tasks/Common/npm-common/npmrcparser.ts

```diff
@@ -9,8 +9,7 @@ export function GetRegistries(npmrc: string): string[] {
     let config = ini.parse(fs.readFileSync(npmrc).toString());
 
     for (let key in config) {
-        let colonIndex = key.indexOf(':');
-        if (key.substring(colonIndex + 1).toLowerCase() === 'registry') {
+        if (key.toLowerCase().endsWith('registry')){
```

Comment:
Why is this change needed? With the proposed change this would include other properties in the config file such as fooRegistry=NotARegistry

Reply:
Reverted.